cluster/ampagent/cmd/install.go

```diff
@@ -326,43 +327,120 @@ func createInitialNetworks() error {
 func removeInitialNetworks() error {
 	for _, network := range ampnetworks {
 		// Check if network already exists
-		exists, err := Docker.NetworkExists(network)
+		id, err := Docker.NetworkID(network)
 		if err != nil {
 			return err
 		}
-		if !exists {
-			log.Println("Skipping non existing network:", network)
-			return nil
+		if id == "" {
+			// Skipping non existent network
+			continue
 		}
 
 		// Remove network
-		time.Sleep(2 * time.Second)
-		if err := Docker.RemoveNetwork(network); err != nil {
+		if err := Docker.RemoveNetwork(id); err != nil {
 			return err
 		}
-		log.Println("Successfully removed network:", network)
+		log.Printf("Successfully removed network %s [%s]", network, id)
+	}
+	return nil
+}
+
+func removeExitedContainers(timeout int) error {
+	i := 0
+	dontKill := []string{"amp-agent", "amp-local"}
+	var containers []types.Container
+	if timeout == 0 {
+		// default value
+		timeout = 30
+	}
+	log.Println("waiting for all services to clear up...")
+	filter := filters.NewArgs()
+	filter.Add("is-task", "true")
+	filter.Add("label", "io.amp.role=infrastructure")
+	for i < timeout {
+		containers, err := Docker.GetClient().ContainerList(context.Background(), types.ContainerListOptions{All: true, Filters: filter})
+		if err != nil {
+			return err
+		}
+		if len(containers) == 0 {
+			log.Println("cleared up")
+			break
+		}
+		for _, c := range containers {
+			switch c.State {
+			case "exited":
+				log.Printf("Removing container %s [%s]\n", c.Names[0], c.Status)
+				err := Docker.GetClient().ContainerRemove(context.Background(), c.ID, types.ContainerRemoveOptions{})
+				if err != nil {
+					if strings.Contains(err.Error(), "already in progress") {
+						// leave it to Docker
+						continue
+					} else {
+						return err
+					}
+				}
+			case "removing":
```

Comment:
In order to group the same code for the same case, you separate the case with commas:
`case "removing", "running":`
It doesn't really matter in this case since both cases are empty, but it's easier to read.